### PR TITLE
add shareable address from default name, leverage for steam join/invite

### PR DESCRIPTION
**Pull request by birarda**
- attempt to retrieve a shareable address for steam join/invite even if connecting via IP or network address

### CODE REVIEW

**Comment by birarda** [comment]
#### Test Plan

Though this functionality was changed for Steam join/invite, you can test it with the "Copy Address" option in the Menus. You'll need to enable advanced settings to see it under the Navigate menu.

From master, go to `hq.highfidelity.io` in your Interface address bar. Use "Copy Address" to confirm that the address begins with `hifi://hq.highfidelity.io`.

From this PR, go to `hq.highfidelity.io` in your Interface address bar. User "Copy Address" to confirm that the address begins with `hifi://hq`.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 43b8e01d46636651a9984b2dcca35bf4eb7954a4 available at: https://gist.github.com/f5ef58cc2b528702dfedd0a61f8ddad3<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8512-43b8e01.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8512-43b8e01.dmg

**Comment by OmegaHeron** [comment]
Testing.

File: interface/src/DiscoverabilityManager.cpp [hunk 1]
```diff
@@ -111,7 +111,7 @@ void DiscoverabilityManager::updateLocation() {
     }
 
     // Update Steam
-    SteamClient::updateLocation(domainHandler.getHostname(), addressManager->currentFacingAddress());
+    SteamClient::updateLocation(domainHandler.getHostname(), addressManager->currentShareableAddress());
```
Comment (Atlante45) [inline interface/src/DiscoverabilityManager.cpp]: This should be `currentFacingShareableAddress`

Comment (birarda) [inline interface/src/DiscoverabilityManager.cpp]: You're right, thanks

**Comment by OmegaHeron** [comment]
Testing Windows 10

Current DEV (5299)
hifi://hq.highfidelity.io/0,0,0/0,0,0,1
hifi://192.168.0.3/1101.79,460.186,-78.4844/0,-0.496747,0,0.867895
hifi://obfuscated.o2tech.info/0,0,0/0,0,0,1

This PR
hifi://hq/0,0,0/0,0,0,1
hifi://o2tech/1101.79,460.186,-78.4844/0,-0.496747,0,0.867895
hifi://heron/0.0717686,0.123688,0.0932733/0,0,0,1

**Comment by hifi-gustavo** [comment]
Build console output for commit: 9461ca0cfca81d4abd088e6b72544ba86300a44b available at: https://gist.github.com/f5ef58cc2b528702dfedd0a61f8ddad3<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8512-9461ca0.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8512-9461ca0.dmg

**Comment by OmegaHeron** [comment]
Testing (latest build) Windows 10

Current DEV (5299)
hifi://hq.highfidelity.io/0,0,0/0,0,0,1
hifi://192.168.0.3/1101.79,460.186,-78.4844/0,-0.496747,0,0.867895
hifi://obfuscated.o2tech.info/0,0,0/0,0,0,1

This PR
hifi://hq/1.81887e-06,-8.04279,8.87364e-06/0,0,0,1
hifi://o2tech/1101.79,460.186,-78.4844/0,-0.496747,0,0.867895
hifi://heron/0.104814,0.123695,0.126277/0,0,0,1

**Pull request merged**